src/RouteGenerator.js

```diff
@@ -0,0 +1,18 @@
+import * as Immutable from 'immutable';
+
+export default class RouteGenerator {
+  constructor() {
+    this._routeToIndexMap = new Immutable.Map();
```

Comment:
Why `Immutable.Map` rather than e.g. standard ES6 `Map` from `babel-runtime` or something? It doesn't look like this is really taking advantage of the immutability at all.